File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/block/BlockSetType.java
package pize.tests.voxelgame.main.block;

public enum BlockSetType{

    DESTROY,
    PLACE,
    REPLACE;

    public static BlockSetType from(boolean oldBlockIsEmpty, boolean newBlockIsEmpty){
        if(oldBlockIsEmpty && !newBlockIsEmpty)
            return PLACE;
        else if(newBlockIsEmpty && !oldBlockIsEmpty)
            return DESTROY;
        return REPLACE;
    }

    public static BlockSetType from(short oldBlockState, short newBlockState){
        return from(
            BlockData.getID(oldBlockState) == 0,
            BlockData.getID(newBlockState) == 0
        );
    }

    public static BlockSetType from(byte oldBlockID, byte newBlockID){
        return from(
            oldBlockID == 0,
            newBlockID == 0
        );
    }

}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/block/BlockSetType.kt
package pize.tests.voxelgame.main.block

enum class BlockSetType {
    DESTROY,
    PLACE,
    REPLACE;

    companion object {
        fun from(oldBlockIsEmpty: Boolean, newBlockIsEmpty: Boolean): BlockSetType {
            if (oldBlockIsEmpty && !newBlockIsEmpty) return PLACE else if (newBlockIsEmpty && !oldBlockIsEmpty) return DESTROY
            return REPLACE
        }

        fun from(oldBlockState: Short, newBlockState: Short): BlockSetType {
            return from(
                BlockData.getID(oldBlockState).toInt() == 0,
                BlockData.getID(newBlockState).toInt() == 0
            )
        }

        fun from(oldBlockID: Byte, newBlockID: Byte): BlockSetType {
            return from(
                oldBlockID.toInt() == 0,
                newBlockID.toInt() == 0
            )
        }
    }
}
